src/chart/draw/consolidateData/transformData.js

```diff
@@ -14,108 +15,51 @@ import { time, set, extent, merge } from 'd3';
 
 export default function transformData(raw, mark) {
     //convenience mappings
-    let config = this.config;
-    let x_behavior = config.x.behavior || 'raw';
-    let y_behavior = config.y.behavior || 'raw';
-    let sublevel = mark.type === 'line'
+    const config = this.config;
+    const x_behavior = config.x.behavior || 'raw';
+    const y_behavior = config.y.behavior || 'raw';
+    const sublevel = mark.type === 'line'
         ? config.x.column
         : mark.type === 'bar' && mark.split ? mark.split : null;
-    let dateConvert = time.format(config.date_format);
-    let totalOrder;
 
     //////////////////////////////////////////////////////////////////////////////////
     // DATA PREP
     // prepare data based on the properties of the mark - drop missing records, etc
     //////////////////////////////////////////////////////////////////////////////////
-
-    // only use data for the current mark
-    raw = mark.per && mark.per.length ? raw.filter(f => f[mark.per[0]] !== undefined) : raw;
-
-    // Make sure data has x and y values
-    if (config.x.column) {
-        raw = raw.filter(f => f[config.x.column] !== undefined);
-    }
-    if (config.y.column) {
-        raw = raw.filter(f => f[config.y.column] !== undefined);
-    }
-
-    //check that x and y have the correct formats
-    if (config.x.type === 'time') {
-        raw = raw.filter(
-            f =>
-                f[config.x.column] instanceof Date
-                    ? f[config.x.column]
-                    : dateConvert.parse(f[config.x.column])
-        );
-        raw.forEach(
-            e =>
-                (e[config.x.column] = e[config.x.column] instanceof Date
-                    ? e[config.x.column]
-                    : dateConvert.parse(e[config.x.column]))
-        );
-    }
-    if (config.y.type === 'time') {
-        raw = raw.filter(
-            f =>
-                f[config.y.column] instanceof Date
-                    ? f[config.y.column]
-                    : dateConvert.parse(f[config.y.column])
-        );
-        raw.forEach(
-            e =>
-                (e[config.y.column] = e[config.y.column] instanceof Date
-                    ? e[config.y.column]
-                    : dateConvert.parse(e[config.y.column]))
-        );
-    }
-
-    if ((config.x.type === 'linear' || config.x.type === 'log') && config.x.column) {
-        raw = raw.filter(f => {
-            return mark.summarizeX !== 'count' && mark.summarizeX !== 'percent'
-                ? +f[config.x.column] || +f[config.x.column] === 0
-                : f;
-        });
-    }
-    if ((config.y.type === 'linear' || config.y.type === 'log') && config.y.column) {
-        raw = raw.filter(f => {
-            return mark.summarizeY !== 'count' && mark.summarizeY !== 'percent'
-                ? +f[config.y.column] || +f[config.y.column] === 0
-                : f;
-        });
-    }
+    const cleaned = cleanData.call(this, mark, raw);
```

Comment:
Careful with all the refactoring - this code is going to be understandable soon! 😆